redisinsight/ui/src/components/live-time-recommendations/LiveTimeRecommendations.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useEffect } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { useHistory } from 'react-router-dom'
+import {
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiFlyoutFooter,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiTitle,
+} from '@elastic/eui'
+import cx from 'classnames'
+
+import { Pages } from 'uiSrc/constants'
+import {
+  recommendationsSelector,
+  fetchRecommendationsAction,
+  readRecommendationsAction,
+  setIsContentVisible
+} from 'uiSrc/slices/recommendations/recommendations'
+import { connectedInstanceSelector } from 'uiSrc/slices/instances/instances'
+import { sendEventTelemetry, TelemetryEvent } from 'uiSrc/telemetry'
+import { ReactComponent as AnalysisIcon } from 'uiSrc/assets/img/icons/analysis.svg'
+import { ReactComponent as DefaultIcon } from 'uiSrc/assets/img/icons/live-time-recommendations.svg'
+import { ReactComponent as HighlightedIcon } from 'uiSrc/assets/img/icons/live-time-recommendations-active.svg'
+
+import Recommendation from './components/recommendation'
+import WelcomeScreen from './components/welcome-screen'
+import styles from './styles.module.scss'
+
+const TIMEOUT_TO_GET_RECOMMENDATION = 10_000
+
+const LiveTimeRecommendations = () => {
+  let interval: NodeJS.Timeout
+  const { id: connectedInstanceId = '', } = useSelector(connectedInstanceSelector)
+  const {
+    data: { recommendations, totalUnread },
+    isContentVisible,
+    isHighlighted
+  } = useSelector(recommendationsSelector)
+
+  const dispatch = useDispatch()
+  const history = useHistory()
+
+  const toggleContent = () => {
+    sendEventTelemetry({
+      event: isContentVisible
+        ? TelemetryEvent.INSIGHTS_RECOMMENDATIONS_CLOSED
+        : TelemetryEvent.INSIGHTS_RECOMMENDATIONS_OPENED,
+      eventData: {
+        databaseId: connectedInstanceId,
+        total: recommendations?.length,
+        list: recommendations.map(({ name }) => name),
```

Comment:
it is a bit strange

```
total: recommendations?.length,
list: recommendations.map(({ name }) => name),
```
You wan't to fail on ``recommendations?.length``, but do not cover ``.map`` method

Reply:
or maybe delete question mark, because you do not check nullish in other places